x-pack/plugins/data_enhanced/public/search/sessions_mgmt/lib/api.ts

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { NotificationsStart } from 'kibana/public';
+import moment from 'moment';
+import { SharePluginStart } from 'src/plugins/share/public';
+import { ISessionsClient } from '../../../../../../../src/plugins/data/public';
+import { BackgroundSessionSavedObjectAttributes } from '../../../../common';
+import {
+  ACTION,
+  EXPIRES_SOON_IN_DAYS,
+  SESSIONS_LISTING_SEARCH_SIZE,
+  STATUS,
+  UISession,
+} from '../../../../common/search/sessions_mgmt';
+
+type UrlGeneratorsStart = SharePluginStart['urlGenerators'];
+
+interface BackgroundSessionSavedObject {
+  id: string;
+  attributes: BackgroundSessionSavedObjectAttributes;
+}
+
+// Helper: factory for a function to map server objects to UI objects
+const mapToUISession = (urls: UrlGeneratorsStart) => async (
+  savedObject: BackgroundSessionSavedObject
+): Promise<UISession> => {
+  // Actions: always allow delete
+  const actions = [ACTION.DELETE];
+
+  const {
+    name,
+    appId,
+    created,
+    expires,
+    status,
+    urlGeneratorId,
+    restoreState,
+  } = savedObject.attributes;
+
+  // calculate expiresSoon flag
+  let expiresSoon = false;
```

Comment:
Since `expiresSoon` is really a computed property, I don't think we need to set both `expires` and `expiresSoon` in the props. Could we extract this into a function and then simply use it when rendering? (Or was the intent to make it easier to test here?)

Reply:
I thought "expires soon" might be a filterable property, but it isn't. You are right, it should be eval'd during render.